kusto/kcsb.go

```diff
@@ -11,191 +10,351 @@ import (
+	managedServiceIdentity           string = "ManagedServiceIdentity"
+	azCli                            string = "AZCLI"
+	interactiveLogin                 string = "InteractiveLogin"
+	domainHint                       string = "RedirectURL"
 )
 
-// authtype mapping
-const (
-	envAuth        string = "EnvironmentVars"
-	azCliAuth      string = "AzCLI"
-	managedIDAuth  string = "ManagedIdentity"
-	clientCredAuth string = "ClientCredentials"
-	appCertAuth    string = "ApplicationCetrifiate"
-	unamePassAuth  string = "UsernamePassword"
-	appTokenAuth   string = "ApplicationToken"
-)
+func assertIfEmpty(key string, value string) {
+	if isEmpty(value) {
+		panic(fmt.Sprintf("Error: %s cannot be null", key))
+	}
+}
+
+func contains(list []string, tofind string) bool {
+	for _, s := range list {
+		if tofind == s {
+			return true
+		}
+	}
+	return false
+}
+
+func assignValue(kcsb *connectionStringBuilder, rawKey string, value string) {
+	rawKey = strings.ToLower(strings.Trim(rawKey, " "))
+	if contains([]string{"datasource", "data source", "addr", "address", "network address", "server"}, rawKey) {
+		kcsb.DataSource = value
+	}
+	if contains([]string{"aad user id", "aaduserid"}, rawKey) {
+		kcsb.AADUserID = value
+	}
+	if contains([]string{"password", "pwd"}, rawKey) {
+		kcsb.Password = value
+	}
+	if contains([]string{"application client id", "applicationclientid", "appclientid"}, rawKey) {
+		kcsb.ApplicationClientId = value
+	}
+	if contains([]string{"application key", "applicationkey", "appkey"}, rawKey) {
+		kcsb.ApplicationKey = value
+	}
+	if contains([]string{"application certificate", "applicationcertificate"}, rawKey) {
+		kcsb.ApplicationCertificate = value
+	}
+	if contains([]string{"application certificate thumbprint", "applicationcertificatethumbprint"}, rawKey) {
+		kcsb.ApplicationCertificateThumbprint = value
+	}
+	if contains([]string{"sendcertificatechain", "send certificate chain"}, rawKey) {
+		kcsb.SendCertificateChain = getBoolValue(value)
+	}
+	if contains([]string{"authority id", "authorityid", "authority", "tenantid", "tenant", "tid"}, rawKey) {
+		kcsb.AuthorityId = value
+	}
+	if contains([]string{"application token", "applicationtoken", "apptoken"}, rawKey) {
+		kcsb.ApplicationToken = value
+	}
+	if contains([]string{"user token", "usertoken", "usrtoken"}, rawKey) {
+		kcsb.UserToken = value
+	}
+	if contains([]string{"msi_auth"}, rawKey) {
+		kcsb.MSIAuthentication = getBoolValue(value)
+	}
+	if contains([]string{"managedserviceidentity", "managed service identity"}, rawKey) {
+		kcsb.ManagedServiceIdentity = value
+	}
+	if contains([]string{"az cli", "azcli"}, rawKey) {
+		kcsb.AZCLI = getBoolValue(value)
+	}
+	if contains([]string{"interactive login", "interactivelogin"}, rawKey) {
+		kcsb.InteractiveLogin = getBoolValue(value)
+	}
+	if contains([]string{"domain hint", "domainhint"}, rawKey) {
+		kcsb.RedirectURL = value
+	}
+}
+
+func getBoolValue(val string) bool {
```

Comment:
equivalent of         boolValue, err := strconv.ParseBool(v) ? Do we need a method for this ?